### PR TITLE
Handling pydantic values, which may not have fields that are expected…

### PR DESCRIPTION
**Pull request by elainethale**
… if a ValidationError has already been hit.

## Description

Systematically went through and made sure all uses of pydantic values should be safe from validation errors. (Otherwise will get KeyError failures instead of seeing the original issues.)

Did not fix the instances that Lixi fixed during the working session.

## Checklist

- [ ] Tests exercising the new feature or bug fix
- [ ] All tests pass
- [ ] At least one code review approval
- [ ] Consider transferring TODOs to JIRA or GitHub

### CODE REVIEW

**Review by daniel-thom — APPROVED** [review]

File: dsgrid/config/dimensions.py [hunk 1]
```diff
@@ -401,6 +406,7 @@ def check_frequency(cls, values):
 
     @validator("ranges", pre=True)
     def check_times(cls, ranges, values):
+        # TODO: Should this be a root_validator with pre=False, skip_on_failure=True?
```
Comment (elainethale) [inline dsgrid/config/dimensions.py:409]: Delete pre and then check that values contains the required fields. Same for below.

File: dsgrid/config/project_config.py [hunk 2]
```diff
@@ -107,6 +107,7 @@ def pre_check_values(cls, values: dict) -> dict:
         check that all from_keys have a match in the to_keys json
 
         """
+        # TODO: Can we delete this?
```
Comment (elainethale) [inline dsgrid/config/project_config.py:110]: Yes

File: dsgrid/submit/submit_dataset.py [hunk 3]
```diff
@@ -102,6 +102,8 @@ def check_for_project_config_expectations(cls, values):
         Check dataset fields to project fields.
         Fields to check: id, model_name, model, sectors
         """
+        # TODO: How are the project_version and teh dataset_version enough to
+        #   pull configs? Where are the project and dataset ids?
```
Comment (elainethale) [inline dsgrid/submit/submit_dataset.py:106]: Delete the file

**Comment by elainethale** [comment]
@daniel-thom I think I might have messed up the dsgrid-test-data submodule? Suggestions?

**Review by lixiliu — APPROVED** [review]
The dsgrid-test-data branch error was probably my fault. Sorry.

**Pull request merged**